### PR TITLE
Merchants will tell when they will restock

### PR DESCRIPTION
**Pull request by KeremBabaG**
#### Summary
Features "Merchants will tell when they will restock"

#### Purpose of change
Adressses #68132. Checking on merchants can be exhausting especially if there is a long distance between you and the merchant. This aims to communicate to the player when the merchants restock.

#### Describe the solution
I created a new parse_tag snippet called <\interval> that gives you the remaining time before they restock. This can be used like "Stick around for <\interval>, a new shipment is on the way" or yada yada.

As of publishing this it is very half baked and gives you a dumb value in negative few thousand seconds, but I aim to make it so that specific time durations will give player different outputs that sounds more daily like, i.E: between 3 or 5 days will be "a few days", under a day will be "soon" etc. Or simply fix it to say a more comprehensible value like "x days x hours"

I put this out in WIP beacuse I wanted to know what everyone thinks

#### Describe alternatives you've considered
cut my heart into a thousand pieces, this is my last resort

Uhh I mean any alternative is welcome; like I said, it is extremely half-baked rn

#### Testing

Replaced Smokes' after trade dialogue with "Come back in <\interval>. I may have something you're interested in. " It worked but put back something like - 435345435 seconds. Working on it as stated above.

#### Additional context
My C++ is extremely undergrad so any feedback is very welcome

### CODE REVIEW

**Comment by KeremBabaG** [comment]
Alright tended to that minus second thing, turns out I misplaced restock and calendar

I'm profoundly stupid

File: data/json/npcs/refugee_center/surface_staff/Smokes/free_merchant_shopkeep_talk.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
   {
     "id": "TALK_FREE_MERCHANTS_MERCHANT_DoneTrading",
     "type": "talk_topic",
-    "dynamic_line": [ "Hope it helps.", "Give me second to write that down… okay, done.", "You know where to find me." ],
+    "dynamic_line": [ "Come back in <interval>. I may have something you're interested in. " ],
```
Comment (casswedson) [inline data/json/npcs/refugee_center/surface_staff/Smokes/free_merchant_shopkeep_talk.json:46]:
```suggestion
    "dynamic_line": [ "Come back in <interval>.  I may have something you're interested in." ],
```

**Comment by KeremBabaG** [comment]
brother I could've sworn that the builds failed without those parameters but whatever I guess

**Review by RenechCDDA — APPROVED** [review]
I think they should have a new talk topic added per-merchant (obviously you do not need to fill in *all* the merchants yourself! Just one as an example) which handles this, but the current approach of changing the 'goodbye' dialogue is fine. (There may be cases where a merchant should not have a player-facing restock timer, but both approaches allow json to handle whether that happens)

The code looks fine, too. Your first draft mixed up `calendar::turn` and `restock` but you already caught that.

**Comment by KeremBabaG** [comment]
> I think they should have a new talk topic added per-merchant (obviously you do not need to fill in _all_ the merchants yourself! Just one as an example) which handles this, but the current approach of changing the 'goodbye' dialogue is fine. (There may be cases where a merchant should not have a player-facing restock timer, but both approaches allow json to handle whether that happens)
> 
> The code looks fine, too. Your first draft mixed up `calendar::turn` and `restock` but you already caught that.

Thanks for the Input!

Do you mean the seperate topic as an option to ask them about restocks? Because that may be better. Also I am thinking of adding them all to merchants, changed only Smokes' dialogue for the test. Will add to the documentation as well.

**Comment by RenechCDDA** [comment]
Yes, I mean a new specific option to ask about the restock.

File: src/npc.cpp [hunk 2]
```diff
@@ -2175,6 +2175,14 @@ void npc::shop_restock()
     distribute_items_to_npc_zones( ret, *this );
 }
 
+std::string npc::get_interval() const
```
Comment (Leonkoithara) [inline src/npc.cpp:2178]: Hi can you change the function name to get_restock_interval()

Comment (Leonkoithara) [reply src/npc.cpp:2178]: Also very cool change can't wait for it to get merged!

Comment (KeremBabaG) [reply src/npc.cpp:2178]: aye thanks, tending to it rn

**Comment by KeremBabaG** [comment]
changed the function name to get_restock_interval

added json dialogue to isherwoods, scrapper, chemist, kostas from godco, artisans, smokes, exodii, hub 01 intercom

**Review by MNG-cataclysm — APPROVED** [review]
Everything looks pretty good to me.

**Pull request merged**